enumerables.rb

```diff
@@ -0,0 +1,137 @@
+module Enumerable
+  def my_each
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i])
+        i += 1
+      end
+    else
+      arr
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i], i)
+        i += 1
+      end
+    else
+      arr
+    end
+  end
+
+  def my_select
+    i = 0
+    to_change = []
+    arr = self
+    if block_given?
+      while i < arr.count
+        to_change << arr[i] if yield(arr[i])
+        i += 1
+      end
+      to_change
+    else
+      arr
+    end
+  end
+
+  def my_any?(arg = nil)
+    arr = self
+    i = 0
+    to_change = false
+    while i < arr.count
+      if !block_given?
+        to_change = true if include?(true) && arg.nil?
+        to_change = true if arr[i] == arg && arg.respond_to?(:to_i)
+        to_change = true if arr[i].match(arg) && arg.is_a?(Regexp)
+        to_change = true if arr[i].instance_of?(arg) && arg.respond_to?(:class)
+      else
+        to_change = false unless yield(arr[i])
+      end
+      i += 1
+    end
+    to_change
+  end
+
+  def my_all?
+    i = 0
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false unless yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
+
+  def my_none?
+    i = 0
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false if yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
+
+  def my_map(the_proc)
+    i = 0
+    to_change = []
+    arr = self
+    arg = the_proc.nil?
+    while i < arr.count
+      to_change << (the_proc.call arr[i]) unless arg == true
+      to_change << yield(arr[i]) if arg == true
+      i += 1
+    end
+    to_change
+  end
+
+  def my_count
+    arr = self
+    i = 0
+    total = 0
+    while i < arr.count
+      arr.my_each { total += 1 }
```

Comment:
- [ ] Please don't use the original #count method to make your my_count.
You can use something simple like length, size, or reuse your own my_each or my_each_with_index methods.

- [ ] my_count should also accept an argument and count the number of times it shows up in the self.

- [ ] my_count should also accept a block and in that case count the number of elements yielding a true value.